lib/workers/global/limits.spec.ts

```diff
@@ -1,21 +1,43 @@
-import { getLimitRemaining, incrementLimit, init } from './limits';
+import {
+  Limit,
+  incLimitedValue,
+  isLimitReached,
+  reset,
+  setMaxLimit,
+} from './limits';
 
 describe('lib/workers/global/limits', () => {
-  describe('init()', () => {
-    it('check defined variables have a value set to zero', () => {
-      const config = { prCommitsPerRunLimit: 3 };
-      init(config);
-      const result = getLimitRemaining('prCommitsPerRunLimit');
-      expect(result).toEqual(3);
-    });
+  beforeEach(() => {
+    reset();
   });
-  describe('incrementLimit()', () => {
-    it('check increment works as expected', () => {
-      const config = { prCommitsPerRunLimit: 3 };
-      init(config);
-      incrementLimit('prCommitsPerRunLimit', 2);
-      const result = getLimitRemaining('prCommitsPerRunLimit');
-      expect(result).toEqual(1);
-    });
+
+  beforeEach(() => {
+    reset();
+  });
+
+  it('increments limited value', () => {
+    setMaxLimit(Limit.Commits, 3);
+
+    expect(isLimitReached(Limit.Commits)).toBe(false);
+
+    incLimitedValue(Limit.Commits, 2);
+    expect(isLimitReached(Limit.Commits)).toBe(false);
+
+    incLimitedValue(Limit.Commits);
+    expect(isLimitReached(Limit.Commits)).toBe(true);
+
+    incLimitedValue(Limit.Commits);
+    expect(isLimitReached(Limit.Commits)).toBe(true);
+  });
+
+  it('defaults to unlimited', () => {
+    expect(isLimitReached(Limit.Commits)).toBe(false);
+  });
+
+  it('increments undefined', () => {
+    incLimitedValue(Limit.Commits);
+    expect(isLimitReached(Limit.Commits)).toBe(false);
+    setMaxLimit(Limit.Commits, 1);
+    expect(isLimitReached(Limit.Commits)).toBe(true);
```

Comment:
Not necessary the case, but is possible. Not sure, maybe it's better to reset `current` counter to `0` instead.

Reply:
We're now resetting counter after `setMaxLimit()` call, since it's more intuitive behavior